modules/web/src/app/shared/directives/value-cahnged.ts

```diff
@@ -0,0 +1,108 @@
+
+  ngOnInit(): void {
+    if (!this._isEditDialog.isEditDialog) {
+      return;
+    }
+
+    const element = this._el.nativeElement;
+    this._previousValue = this._control.control.value;
+
+    switch (element.classList[0]) {
+      case 'mat-input-element':
+        if (element.getAttribute('type') === 'number') {
+          this._previousValue = !element.getAttribute('ng-reflect-model')
+            ? this._control.control.value
+            : parseInt(element.getAttribute('ng-reflect-model'));
+        }
+        break;
+      case 'mat-checkbox':
+        this._previousValue = this._control.control.value === null ? false : this._control.control.value;
+        break;
+      case 'mat-radio-group':
+        element.classList.toggle('radio-group-column', element.style.flexDirection === 'column');
+        break;
+      default:
+        this._previousValue = this._control.control.value;
+        break;
+    }
+
+    this._control?.control?.valueChanges.subscribe(value => {
+      switch (element.classList[0]) {
+        case 'mat-select':
+          this.addClassToMatSelect(element, value);
+          break;
+        case 'mat-input-element':
+          element.classList.toggle(
+            'value-changed',
+            !!this._previousValue || !!value ? this._previousValue !== value : false
+          );
+          break;
+        case 'mat-radio-group':
+          this._previousValue = !this._previousValue ? value : this._previousValue;
+          element.classList.toggle('value-changed', this._previousValue !== value);
+          break;
+        case 'ngx-monaco-editor':
+          this._previousValue = this._previousValue === null ? value : this._previousValue;
+          element.classList.toggle('value-changed', this._previousValue !== value);
+          break;
+        default:
+          this._previousValue = this._previousValue === null ? value : this._previousValue;
+          element.classList.toggle('value-changed', this._previousValue !== value);
+      }
+    });
+  }
+
+  isArraysEqual(arr1: string[], arr2: string[]): boolean {
+    if (arr1?.length !== arr2?.length) {
+      return false;
+    }
+    for (let i = 0; i < arr1?.length; i++) {
+      if (arr1[i] !== arr2[i]) {
+        return false;
+      }
+    }
+    return true;
+  }
+
+  addClassToMatSelect(element: any, value: any): void {
+    const classList = [...element.classList];
+
+    if (classList?.includes('mat-select-multiple')) {
+      if (this._previousValue && value) {
+        element.classList.toggle('value-changed', !this.isArraysEqual(this._previousValue, value));
+      } else if (!this._previousValue && (!value || !value?.length)) {
+        element.classList.remove('value-changed');
+      } else {
+        element.classList.add('value-changed');
+      }
+    } else {
+      this._previousValue = !this._previousValue ? value : this._previousValue;
```

Comment:
Just curious, since `_previousValue` is initial value and we don't update it for other type of elements so why do we need to update it here?